### DISCUSSION TITLE
[ERROR GRT-0076] Net foo.bar\[13\] not properly covered.

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2925] [General]
What does this error message mean?

**Comment by maliberty** [comment]
@eder-matheus please respond

**Comment by eder-matheus** [comment]
@oharboe That means the guides don't cover at least one pin of this net. I will update this error message to clarify what pin is not covered.
Could you share what commit you are using in this run? This error should not be happening at all and could be a side effect of an update I've made.

**Reply by oharboe** [reply]
I see.

The problem appeared, I think, because the wrong macro placement file was used, so the macros were overlapping.

What error messages you get when the macros are outside of the core area(common user error with configurable size design) or overlapping is a bit hit and miss in my experience.

This was with the most recent master of ORFS and OpenROAD.

**Reply by eder-matheus** [reply]
Could you try this branch with your test case: https://github.com/eder-matheus/OpenROAD/tree/grt_test? Also, do you know the min and max pin layers of this net?

**Reply by oharboe** [reply]
This was with ASAP7, no change in default config.

I am afraid I lost the testcase as it was an incorrect macro layout configuration.

**Reply by eder-matheus** [reply]
Oh, so the error went away after fixing the macro placement? If so, that's good. I will update the error message to be more valuable.

Also, you can use the `make global_route_issue` command to create the tar file with the reproducible test case when you face some error (same for other stages of the flow), so we can keep track of failing cases.

**Reply by oharboe** [reply]
Yes.

Thanks for the tip! Will do.

**Reply by oharboe** [reply]
Though.... I suspect incorrect floorplan is prone to confusing error messages unless an explicit check to humanize the error messages is added:

- warning: not aligning to grid , no halo or too big?
- error: overlapping macros, macros outside core area, too close to edge

etc.

Then again, a glance at the floorplan tells more than a thousand words...

Maybe a "make gui_error" feature to view errors and warnings in the gui might help. I had an idea that warnings/errors could be annotated with location, like DRC errors, would be neat.

**Comment by oharboe** [comment]
I don't recall the details, but this was fixed on master.
